system-contracts/contracts/test-contracts/MsgLogger.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {ISystemContract} from "../interfaces/ISystemContract.sol";
+
+contract MsgLogger is ISystemContract {
+    event Called(uint256 value, bytes data, bool systemFlag);
+
+    fallback() external payable onlySystemCall {
```

Comment:
You can use `SystemContractHelper.isSystemCall` method to get value instead of modifier.